### PR TITLE
Remove rcpputils::fs dependencies from rosbag2_storages

### PR DESCRIPTION
**Pull request by r7vme**
Original author: Kenta Yonekura <yoneken@ieee.org>

Copy of stalled PR https://github.com/ros2/rosbag2/pull/1474

### CODE REVIEW

**Comment by r7vme** [comment]
@MichaelOrlov could you please take a look?

File: rosbag2_storage/src/rosbag2_storage/metadata_io.cpp
Comment (r7vme) [inline rosbag2_storage/src/rosbag2_storage/metadata_io.cpp:58]: @MichaelOrlov I'm not 100% sure that `generic_string` is necessary. This only affects behaviour on Windows

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20197/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14617/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20908/)

File: rosbag2_storage/src/rosbag2_storage/metadata_io.cpp
Comment (clalancette) [reply rosbag2_storage/src/rosbag2_storage/metadata_io.cpp:58]: > @MichaelOrlov I'm not 100% sure that `generic_string` is necessary. This only affects behaviour on Windows

I believe that's why we added it earlier, but we'll see what Windows CI has to say about this code.

**Comment by clalancette** [comment]
@r7vme There are quite a few linter failures here.  Please run the linters locally and fix the problems, then we can try again.

**Comment by r7vme** [comment]
@clalancette fixed tests, PTAL

**Comment by clalancette** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20208/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14628/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20927/)

**Review by clalancette — APPROVED** [review]
This looks good to me, but I'd like another opinion from @MichaelOrlov before we merge.

**Review by MichaelOrlov — APPROVED** [review]
@r7vme Thanks for the contribution. The changes looks good to me.
However, there is one open question for the future.
Do we really need to use the ".generic_string()" or simple `.string()` would be enough?

**Comment by clalancette** [comment]
> Do we really need to use the ".generic_string()" or simple `.string()` would be enough?

Someone is going to have to test it out on Windows to tell for sure.

**Comment by MichaelOrlov** [comment]
As far as I understand from documentation the one should use  `.generic_string()` if needed to make sure that the Windows-specific quirks with backslashes `\` in the path will be converted to the regular slashes `/` when the path needs to be represented as string.

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport iron humble

**Pull request merged**
